### DISCUSSION TITLE
Error compiling

### DISCUSSION THREAD

**Discussion by Matteo8825** [Sleeper85/esphome-jk-bms-can#16] [General]
hello Sleeper85, 
I need your help.
I have a problem during the compail file yaml with command line.
my error is:

WARNING GPIO2 is a strapping PIN and should only be used for I/O with care.
Attaching external pullup/down resistors to strapping pins can cause unexpected failures.
See https://esphome.io/guides/faq.html#why-am-i-getting-a-warning-about-strapping-pins
Failed config

jk_bms_ble: [source <unicode string>:142]

  Component not found: jk_bms_ble.
  - ble_client_id: client0
    protocol_version: JK02_32S
    throttle: 5s
    id: bms0
binary_sensor.jk_bms_ble: [source <unicode string>:151]

  Platform not found: 'binary_sensor.jk_bms_ble'.
  platform: jk_bms_ble
  balancing:
    name: jk-ble-can BMS Balancing
  charging:
    name: jk-ble-can BMS Charging
  discharging:
    name: jk-ble-can BMS Discharging
  online_status:
    name: jk-ble-can Online Status
button.jk_bms_ble: [source <unicode string>:162]

  Platform not found: 'button.jk_bms_ble'.
  platform: jk_bms_ble
  retrieve_settings:
    name: jk-ble-can retrieve settings
  retrieve_device_info:
    name: jk-ble-can retrieve device info
number.jk_bms_ble: [source <unicode string>:169]

  Platform not found: 'number.jk_bms_ble'.
  platform: jk_bms_ble
  jk_bms_ble_id: bms0
  balance_trigger_voltage:
    name: jk-ble-can balance trigger voltage
  cell_count:
    name: jk-ble-can cell count
  total_battery_capacity:
    name: jk-ble-can total battery capacity
  cell_voltage_overvoltage_protection:
    name: jk-ble-can cell voltage overvoltage protection
  cell_voltage_overvoltage_recovery:
    name: jk-ble-can cell voltage overvoltage recovery
  cell_voltage_undervoltage_protection:
    name: jk-ble-can cell voltage undervoltage protection
  cell_voltage_undervoltage_recovery:
    name: jk-ble-can cell voltage undervoltage recovery
  balance_starting_voltage:
    name: jk-ble-can balance starting voltage
  voltage_calibration:
    name: jk-ble-can voltage calibration
  current_calibration:
    name: jk-ble-can current calibration
  power_off_voltage:
    name: jk-ble-can power off voltage
  max_balance_current:
    name: jk-ble-can max balance current
  max_charge_current:
    name: jk-ble-can max charge current
  max_discharge_current:
    name: jk-ble-can max discharge current
sensor.jk_bms_ble: [source <unicode string>:270]

  Platform not found: 'sensor.jk_bms_ble'.
  platform: jk_bms_ble
  jk_bms_ble_id: bms0
  min_cell_voltage:
    id: min_cell_voltage
    name: jk-ble-can min cell voltage
  max_cell_voltage:
    id: max_cell_voltage
    name: jk-ble-can max cell voltage
  min_voltage_cell:
    id: min_voltage_cell
    name: jk-ble-can min voltage cell
  max_voltage_cell:
    id: max_voltage_cell
    name: jk-ble-can max voltage cell
  delta_cell_voltage:
    name: jk-ble-can delta cell voltage
  average_cell_voltage:
    name: jk-ble-can average cell voltage
  cell_voltage_1:
    name: jk-ble-can cell voltage 1
  cell_voltage_2:
    name: jk-ble-can cell voltage 2
  cell_voltage_3:
    name: jk-ble-can cell voltage 3
  cell_voltage_4:
    name: jk-ble-can cell voltage 4
  cell_voltage_5:
    name: jk-ble-can cell voltage 5
  cell_voltage_6:
    name: jk-ble-can cell voltage 6
  cell_voltage_7:
    name: jk-ble-can cell voltage 7
  cell_voltage_8:
    name: jk-ble-can cell voltage 8
  cell_voltage_9:
    name: jk-ble-can cell voltage 9
  cell_voltage_10:
    name: jk-ble-can cell voltage 10
  cell_voltage_11:
    name: jk-ble-can cell voltage 11
  cell_voltage_12:
    name: jk-ble-can cell voltage 12
  cell_voltage_13:
    name: jk-ble-can cell voltage 13
  cell_voltage_14:
    name: jk-ble-can cell voltage 14
  cell_voltage_15:
    name: jk-ble-can cell voltage 15
  cell_voltage_16:
    name: jk-ble-can cell voltage 16
  cell_resistance_1:
    name: jk-ble-can cell resistance 1
  cell_resistance_2:
    name: jk-ble-can cell resistance 2
  cell_resistance_3:
    name: jk-ble-can cell resistance 3
  cell_resistance_4:
    name: jk-ble-can cell resistance 4
  cell_resistance_5:
    name: jk-ble-can cell resistance 5
  cell_resistance_6:
    name: jk-ble-can cell resistance 6
  cell_resistance_7:
    name: jk-ble-can cell resistance 7
  cell_resistance_8:
    name: jk-ble-can cell resistance 8
  cell_resistance_9:
    name: jk-ble-can cell resistance 9
  cell_resistance_10:
    name: jk-ble-can cell resistance 10
  cell_resistance_11:
    name: jk-ble-can cell resistance 11
  cell_resistance_12:
    name: jk-ble-can cell resistance 12
  cell_resistance_13:
    name: jk-ble-can cell resistance 13
  cell_resistance_14:
    name: jk-ble-can cell resistance 14
  cell_resistance_15:
    name: jk-ble-can cell resistance 15
  cell_resistance_16:
    name: jk-ble-can cell resistance 16
  total_voltage:
    id: total_voltage
    name: jk-ble-can total voltage
  current:
    id: current
    name: jk-ble-can current
  power:
    name: jk-ble-can power
  charging_power:
    name: jk-ble-can charging power
  discharging_power:
    name: jk-ble-can discharging power
  temperature_sensor_1:
    id: temperature_sensor_1
    name: jk-ble-can temperature sensor 1
  temperature_sensor_2:
    id: temperature_sensor_2
    name: jk-ble-can temperature sensor 2
  power_tube_temperature:
    id: power_tube_temperature
    name: jk-ble-can power tube temperature
  state_of_charge:
    id: state_of_charge
    name: jk-ble-can state of charge
  capacity_remaining:
    name: jk-ble-can capacity remaining
  total_battery_capacity_setting:
    id: total_battery_capacity_setting
    name: jk-ble-can total battery capacity setting
  charging_cycles:
    id: charging_cycles
    name: jk-ble-can charging cycles
  total_charging_cycle_capacity:
    name: jk-ble-can total charging cycle capacity
  total_runtime:
    name: jk-ble-can total runtime
  balancing_current:
    name: jk-ble-can balancing current
  errors_bitmask:
    id: errors_bitmask
    name: jk-ble-can errors bitmask
switch.jk_bms_ble: [source <unicode string>:452]

  Platform not found: 'switch.jk_bms_ble'.
  platform: jk_bms_ble
  charging:
    id: charging_switch
    name: jk-ble-can BMS Charge switch
  discharging:
    id: discharging_switch
    name: jk-ble-can BMS Discharge switch
  balancer:
    name: jk-ble-can BMS Balance switch
text_sensor.jk_bms_ble: [source <unicode string>:485]

  Platform not found: 'text_sensor.jk_bms_ble'.
  platform: jk_bms_ble
  errors:
    name: jk-ble-can errors
  total_runtime_formatted:
    name: jk-ble-can total runtime formatted

**Comment by Sleeper85** [comment]
Hi,

jk_bms_ble is an external component. Does the PC from which you compiled the YAML have access to the Internet?

**Reply by Matteo8825** [reply]
yes, I had deleted the part of the external components code, now I have reinserted it, but it gives me the following error:

INFO ESPHome 2023.12.7
INFO Reading configuration jk_ble_can.yaml...
INFO Cloning https://github.com/syssi/esphome-jk-bms.git@main
Failed config

external_components: [source <unicode string>:55]
  -
    git is not installed but required for external_components.
    Please see https://git-scm.com/book/en/v2/Getting-Started-Installing-Git for installing git.
    source: github://syssi/esphome-jk-bms@main
    refresh: 0s

[jk-ble-can.txt](https://github.com/Sleeper85/esphome-jk-bms-can/files/14230388/jk-ble-can.txt)

**Reply by Sleeper85** [reply]
You need 'git' to clone the repository but it is not installed.

I can already tell you that the BLE version is not stable with Atom Lite.

The BLE version is stable with Atom S3. I need to update the readme.

**Reply by Matteo8825** [reply]
Since I'm a beginner, can you tell me how to install "git"?
since I use atom lite then I will try with the wire version

**Reply by Matteo8825** [reply]
I did it, thank you very much for the support
